x/proof/keeper/msg_server_submit_proof.go

```diff
@@ -62,6 +62,8 @@ func (k msgServer) SubmitProof(ctx context.Context, msg *types.MsgSubmitProof) (
 		ClosestMerkleProof: msg.Proof,
 	}
 
+	// TODO_IMPROVE(@h5law #427): Utilise smt.VerifyCompactClosestProof here to
+	// reduce on-chain storage requirements for proofs.
```

Comment:
I don't think this comment is in quite the right place. Ideally, this PR would be based on #406 (or #417, which is downstream from there). Closest Merkle proof verification isn't done in `queryAndValidateClaimForProof`.

![image](https://github.com/pokt-network/poktroll/assets/600733/546ae26b-861b-4e2a-88ea-8024c7e8efc3)

![image](https://github.com/pokt-network/poktroll/assets/600733/3074c5b7-2e2c-472b-bce5-f837a8bbfec1)

Reply:
👍